core/src/test/java/org/apache/iceberg/actions/TestSortStrategy.java

```diff
@@ -0,0 +1,142 @@
+
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.MockFileScanTask;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableTestBase;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestSortStrategy extends TableTestBase {
+
+  @Parameterized.Parameters(name = "formatVersion = {0}")
+  public static Object[] parameters() {
+    return new Object[] {2}; // We don't actually use the format version since everything is mock
+  }
+
+  @Override
+  public void setupTable() throws Exception {
+    super.setupTable();
+    table.replaceSortOrder().asc("data").commit();
+  }
+
+  private static final long MB = 1024 * 1024;
+
+  public TestSortStrategy(int formatVersion) {
+    super(formatVersion);
+  }
+
+  class TestSortStrategyImpl extends SortStrategy {
+
+    @Override
+    public Table table() {
+      return table;
+    }
+
+    @Override
+    public Set<DataFile> rewriteFiles(List<FileScanTask> filesToRewrite) {
+      throw new UnsupportedOperationException();
+    }
+  }
+
+  private SortStrategy defaultSort() {
+    return (SortStrategy) new TestSortStrategyImpl().options(Collections.emptyMap());
+  }
+
+  private List<FileScanTask> tasksForSortOrder(int sortOrderId, int... fileSizesMB) {
+    ImmutableList.Builder<FileScanTask> files = ImmutableList.builder();
+    IntStream.of(fileSizesMB).forEach(length -> files.add(MockFileScanTask.mockTask(length * MB, sortOrderId)));
+    return files.build();
+  }
+
+  @Test
+  public void testInvalidSortOrder() {
+    AssertHelpers.assertThrows("Should not allow an unsorted Sort order", IllegalArgumentException.class,
+        () -> defaultSort().sortOrder(SortOrder.unsorted()).options(Collections.emptyMap()));
+
+    AssertHelpers.assertThrows("Should not allow a Sort order with bad columns", ValidationException.class,
+        () -> {
+          Schema badSchema = new Schema(
+              ImmutableList.of(Types.NestedField.required(0, "nonexistant", Types.IntegerType.get())));
+
+          defaultSort()
+              .sortOrder(SortOrder.builderFor(badSchema).asc("nonexistant").build())
+              .options(Collections.emptyMap());
+        });
+  }
+
+  @Test
+  public void testSelectAll() {
+    List<FileScanTask> invalid = ImmutableList.<FileScanTask>builder()
+        .addAll(tasksForSortOrder(-1, 500, 500, 500, 500))
```

Comment:
what is order id -1? If it's default UNSORTED_ORDER, it should be 0.

Reply:
Currently it doesn't matter because of the note I wrote below, since we normally do not set this value on our writers